packages/block-library/tsconfig.json

```
@@ -0,0 +1,14 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"declarationDir": "build-types",
+		"types": [ "gutenberg-env" ],
+		"strictNullChecks": true
+	},
+	"references": [ { "path": "../element" } ],
+	"include": [
+		"src/table-of-contents/list.tsx",
+		"src/table-of-contents/utils.ts"
+	]
```

I'd defer to folks who have specific TS-in-Gutenberg experience to weigh in here on the best practice, but every other package opting into TS is using wildcards, with the exception of `docgen`. Seems like the following would be more consistent:

```suggestion
	"include": [ "src/**/*" ]
```

---

Last time I tried that, the types weren't well-defined enough in the block library package (or its dependencies) to enable type checking for the whole package. It caused errors to be reported all over the place.

I still wanted to use TypeScript for these 2 files, though, because the JSDoc versions were significantly more cluttered to read, hence why I did what I did.

That said, I just realized that I could make this more general without causing a bunch of new errors to be reported by simply limiting the included files to those with a TypeScript extension:
```suggestion
	"include": [ "src/**/*.ts", "src/**/*.tsx" ]
```

So I've gone ahead and done just that.